dom.go

```diff
@@ -275,46 +331,99 @@ func (h *HTML) reconcileChildren(prev *HTML, insertBefore *HTML) (pendingMounts
 			if m, ok := nextChild.(Mounter); ok {
 				pendingMounts = append(pendingMounts, m)
 			}
-			if insertBefore != nil {
-				h.node.Call("insertBefore", nextChildRender.node, insertBefore.node)
-				continue
-			}
-			h.node.Call("appendChild", nextChildRender.node)
+			h.insertBefore(h.insertBeforeNode, nextChildRender, nil)
 			continue
 		}
 
-		prevChild := prev.children[i]
+		var prevChild ComponentOrHTML
+		if len(prev.children) > i {
+			prevChild = prev.children[i]
+		}
```

Comment:
This change introduces a lot of uncertainty for me about whether or not at this point in the loop we're dealing with children that have a direct relationship to `prev` or not. It seems this is only the case when `keyed == true`, so I think it'd be more clear to have this `prevChild` declaration be inside the `if` below and then keep the old `prevChild` declaration that you removed but have it defined below this `keyed` if statement.

Reply:
The `if` check here is required, because if we're dealing with keyed children, the [earlier condition](https://github.com/gopherjs/vecty/pull/147/files/1371c31a8088ea66eb45d93dba67c0f75d4aabb0#diff-cf11d578486f49277bfe20d59ea10cecR317) may not have `continue`d the children loop when we get here - we still want to assign this var in either case, it just needs this check now to avoid overflowing the prev.children array when keyed.